devito/ir/clusters/cluster.py

```diff
@@ -343,6 +343,21 @@ def dspace(self):
         intervals = intervals.promote(lambda d: not d.is_Sub)
         intervals = intervals.zero(set(intervals.dimensions) - oobs)
 
+        # Intersect with intervals from buffered dimensions. Unions of
```

Comment:
I'm not quite sure why would do this here, this really seems like a small tweak to `process_arguments`